operator/api/redpanda/v1alpha2/conversion/to_render.go

```diff
@@ -0,0 +1,285 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package conversion
+
+import (
+	applycorev1 "k8s.io/client-go/applyconfigurations/core/v1"
+	"k8s.io/utils/ptr"
+
+	"github.com/redpanda-data/redpanda-operator/charts/redpanda/v25"
+	"github.com/redpanda-data/redpanda-operator/gotohelm/helmette"
+	redpandav1alpha2 "github.com/redpanda-data/redpanda-operator/operator/api/redpanda/v1alpha2"
+	"github.com/redpanda-data/redpanda-operator/pkg/kube"
+)
+
+// V2Defaults contains the default values for the v2 CRD conversion.
+type V2Defaults struct {
+	RedpandaImage    *redpandav1alpha2.RedpandaImage
+	SidecarImage     *redpandav1alpha2.RedpandaImage
+	ConfiguratorArgs []string
+}
+
+// ConvertV2ToRenderState converts a v2 Redpanda CRD to a redpanda chart RenderState.
+func ConvertV2ToRenderState(config *kube.RESTConfig, defaults *V2Defaults, cluster *redpandav1alpha2.Redpanda, _pools any) (*redpanda.RenderState, error) {
+	spec := defaultV2Spec(defaults, cluster)
+
+	dot, err := redpanda.Chart.Dot(config, helmette.Release{
+		Namespace: cluster.Namespace,
+		Name:      cluster.GetHelmReleaseName(),
+		Service:   "Helm",
```

Comment:
nit: consider changing this to `redpanda-operator` or punt, similar to my comment about the chart version label.